dandiapi/api/models/version.py

```diff
@@ -114,27 +114,25 @@ def publish_status(self) -> Version.Status:
         return Version.Status.VALID
 
     @property
-    def publish_validation_error(self) -> str:
-        if self.validation_error:
-            return self.validation_error
-
+    def asset_validation_errors(self) -> list[str]:
         # Import here to avoid dependency cycle
         from .asset import Asset
 
         # Assets that are not VALID (could be pending, validating, or invalid)
-        invalid_assets = self.assets.filter(status=Asset.Status.INVALID).count()
-        if invalid_assets > 0:
-            return f'{invalid_assets} invalid asset metadatas'
+        invalid_assets = self.assets.filter(status=Asset.Status.INVALID)
+        errors = [invalid_asset.validation_error for invalid_asset in invalid_assets]
```

Comment:
```suggestion
        invalid_assets = self.assets.filter(status=Asset.Status.INVALID).values('validation_error')
        errors = [invalid_asset['validation_error'] for invalid_asset in invalid_assets]
```
This way the query only returns the `validation_error` fields. Iterating over every asset implicitly queries for everything else, including metadata.
The same optimization can be done on `unvalidated_assets` below.

Reply:
https://github.com/dandi/dandi-api/pull/417/commits/a1e1435dbe48a781c293b8dc917c8bc3bd0ff146